files/libs/xroad_libs.sh

```diff
@@ -41,189 +41,184 @@ function destroy_api_key () {
+  else
+    log "Warning, Could not create csr for type $key_usage_type. $api_response_body"
   fi
 }
 
 function request_certificate () {
   local type=$1
   local ca_enrollment_endpoint=$2
-  local username=$3
-  local password=$4
-  local id=$(readlink $type|cut -d . -f "1")
+  local username="$type-$PX_MEMBER_CODE"
+  local password=$3
+  local csr_path="/tmp/$type-${PX_MEMBER_CODE}.csr"
+  local crt_path="/tmp/$type-${PX_MEMBER_CODE}.crt"
 
   # request certificate if it doesn't exists
-  if [ ! -f ${id}.crt ]; then
-    log "requesting $type certificate"
+  if [ ! -f $crt_path ]; then
+    log "requesting $type certificate for $username"
     # curl -k allows untrusted HTTPS connection.
     # Flag must be removed before release when EJBCA endpoint is available from public internet with a trusted certificate.
     curl -s -L $ca_enrollment_endpoint \
          --form-string user=$username \
          --form-string password=$password \
          -F "pkcs10req=
-         $(cat ${id}.csr)
+         $(cat ${csr_path})
          " \
-         --form-string 'resulttype=4' > ${id}.crt # resulttype 4 returns full chain
+         --form-string 'resulttype=4' > ${crt_path} # resulttype 4 returns full chain
     # Validate certificate with openssl
-    openssl x509 -noout -in ${id}.crt 2> /dev/null
+    openssl x509 -noout -in ${crt_path} 2> /dev/null
     # Check openssl status code, exit on failure
     check $? "certificate request from certification authority failed"
   else
-    log "$type certificate already requested"
+    log "$type certificate already requested for $username"
   fi
 }
 
-function signer_console () {
-  # we define signer_console function, because singer-console uses rlwrap.
-  # rlwrap fails with "rlwrap: error: My terminal reports width=0" at docker entrypoint
-  . /etc/xroad/services/signer-console.conf
-  java ${SIGNER_CONSOLE_PARAMS} ${XROAD_PARAMS} -jar /usr/share/xroad/jlib/signer-console.jar "$@"
+function import_certificate () {
+  local type=$1
+  local crt_path="/tmp/$type-${PX_MEMBER_CODE}.crt"
+  log "Importing $type certificate for $PX_MEMBER_CODE"
+
+  local api_key=${created_api_key[1]}
+  local curl_args=('-H' 'Content-Type: application/octet-stream' '-H' "Authorization: X-Road-ApiKey token=$api_key")
+  api_response=$(curl -s -k -i -X POST --data-binary "@$crt_path" "${curl_args[@]}" "https://localhost:4000/api/v1/token-certificates")
+  api_response_status_code=$(echo -e "$api_response"|head -n 3|tail -n 1|cut -d$' ' -f2)
+  api_response_body=$(echo -e "$api_response"|tail -n 1)
+
+  if [ $api_response_status_code = 201 ]; then
+    log "Uploaded $type-${PX_MEMBER_CODE}.crt"
+    if [ $type = auth ]; then
+      # register and activate auth cert only
+      crt_hash=$(echo $api_response_body|jq -r '.certificate_details.hash')
+      request_api PUT "/token-certificates/$crt_hash/register" "{\"address\":\"${PX_SS_PUBLIC_ENDPOINT}\"}"
```

Comment:
during setup, this will result
```
{"status":500,"error":{"code":"core.Signer.UnknownMember","metadata":["Could not find any certificates for member 'MEMBER:JP-TEST/COM/0170121212121'. Are you sure tokens containing the certifications are logged in?"]}}
```
but when running the same script after initialization, it works.